use-latest-deps-dotnet.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+print_usage () {
+  (>&2 echo "Usage:")
+  (>&2 echo "    $0 [-d] github-user/repository-name")
+  (>&2 echo "Arguments:")
+  (>&2 echo "    -d: do a dry-run. Don't push or send a PR.")
+}
+
+# Check for optional arguments.
+DRYRUN=0
+while getopts :d opt; do
+  case $opt in
+    d)
+      (>&2 echo "Entered dry-run mode.")
+      DRYRUN=1
+      ;;
+    \?)
+      (>&2 echo "Got invalid option -$OPTARG.")
+      print_usage
+      exit 1
+      ;;
+  esac
+done
+shift $((OPTIND-1))
+
+
+# Check that positional arguments are set.
+if [[ -z $1 ]] ; then
+  (>&2 echo "Missing repo argument.")
+  print_usage
+  exit 1
+fi
+if [[ "$1" != *"/"* ]] ; then
+  (>&2 echo "Repo argument needs to be of form username/repo-name.")
+  print_usage
+  exit 1
+fi
+REPO=$1
+
+
+# Get this script's directory.
+# http://stackoverflow.com/a/246128/101923
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+set -e
+set -x
+
+# Find list of all csproj files, then cycle through them updating version numbers.
+shopt -s globstar
+for i in **/*.csproj; do
+  dotnet outdated "$i" -u --include "Google"
```

Comment:
Is this the general pattern we follow for all repos?  Upgrading only the google packages?

Reply:
No, though the Java version does have an entire rules xml file that has dozens of rules about which packages to update.

When I tested out the script, there were several like testing dependencies that, while an update may not hurt or break anything, could increase the amount of work needed to update it. But I am happy to take this out if you want to update all dependencies, I'll leave that up to your discretion as the current repo maintainer.

Reply:
I agree with the current strategy.  When we feel bolder, later, we may change it to upgrade all packages.